doc/guides/device_mgmt/smp_groups/smp_group_1.rst

```diff
@@ -0,0 +1,401 @@
+    | "rsn"                 | Optional string that clarifies reason for an      |
+    |                       | error; specifically useful for error code ``1``,  |
+    |                       | unknown error.                                    |
+    +-----------------------+---------------------------------------------------+
+
+Set state of image request
+==========================
+
+Set state of image request header fields
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``2``  | ``1``        |  ``0``         |
+    +--------+--------------+----------------+
+
+CBOR data of request command:
+
+
+.. code-block:: none
+
+    {
+        {
+            (str)"hash"         : (str)
+            (str)"confirm"      : (bool)
+        }
+    }
+
+If "confirm" is false an image with the "hash" will be set for test, which means
+that it will not be marked as permanent and upon hard reset the previous
+application will be restored to the primary slot.
+
+Set state of image response
+============================
+
+The response takes the same format as :ref:`mcumgr_smp_protocol_op_1_grp_1_cmd_0`
+
+Image upload
+************
+
+The image upload command allows to update application image.
+
+Image upload request
+====================
+
+The image upload request is sent for each chunk of image that is uploaded, until
+complete image gets uploaded to a device.
+
+Set state of image request header fields
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``2``  | ``1``        |  ``1``         |
+    +--------+--------------+----------------+
+
+CBOR data of request command:
+
+.. code-block:: none
+
+    {
+        {
+            (str,opt)"image"    : (uint)
+            (str,opt)"len"      : (uint)
+            (str)"off"          : (uint)
+            (str,opt)"sha"      : (str)
+            (str,opt)"data"     : (byte str)
+            (str,opt)"upgrade"  : (bool)
+        }
+    }
+
+where
+
+.. table::
+    :align: center
+
+    +-----------------------+---------------------------------------------------+
+    | "image"               | optional image number                             |
+    +-----------------------+---------------------------------------------------+
+    | "len"                 | optional length of image, it only appears in the  |
+    |                       | first packet of request, where "off" is 0         |
+    +-----------------------+---------------------------------------------------+
+    | "off"                 | offset of image chunk the request carries         |
+    +-----------------------+---------------------------------------------------+
+    | "sha"                 | string identifying update session; although name  |
```

Comment:
This should only be present if "off" is = 0.

Reply:
Done